### PR TITLE
Create a separate workflow for benchmark pipeline

### PR DESCRIPTION
**Pull request by pzread**
Move the benchmark-related jobs into a separate workflow and call from `ci.yml`.

The concurrency isn't needed in the reusable workflow as the cancellation on its caller will propagate to it.

### CODE REVIEW

File: .github/workflows/benchmarks.yml [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+        type: string
+      tf-binaries-archive:
+        required: true
+        type: string
+      tf-binaries-gcs-artifact:
+        required: true
+        type: string
+
+jobs:
+  build_benchmark_suites:
```
Comment (GMNGeoffrey) [inline .github/workflows/benchmarks.yml:45]: I think now that the whole workflow is called "benchmarks" we can drop that from the job names

```suggestion
  build_suites:
```

File: .github/workflows/benchmarks.yml [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    runs-on:
+      # Hacks, and order matters. See the comment at the top of the ci.yml.
+      - self-hosted
```
Comment (GMNGeoffrey) [inline .github/workflows/benchmarks.yml:48]: We should be able drop/update this comment
```suggestion
      - self-hosted # must come first
```

File: .github/workflows/benchmarks.yml [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+      BUILD_DIR_ARCHIVE: ${{ inputs.build-dir-archive }}
+      BUILD_DIR_GCS_ARTIFACT: ${{ inputs.build-dir-gcs-artifact }}
+      TF_BINARIES_DIR: ${{ inputs.tf-binaries-dir }}
+      TF_BINARIES_ARCHIVE: ${{ inputs.tf-binaries-archive }}
+      TF_BINARIES_GCS_ARTIFACT: ${{ inputs.tf-binaries-gcs-artifact }}
+    steps:
+      - name: "Checking out repository"
+        uses: actions/checkout@7884fcad6b5d53d10323aee724dc68d8b9096a2e # v2
+        with:
+          submodules: true
```
Comment (GMNGeoffrey) [inline .github/workflows/benchmarks.yml:64]: Do we actually need the submodules?

File: .github/workflows/benchmarks.yml [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+          build_tools/github_actions/docker_run.sh \
+            --env "IREE_TF_BINARIES_DIR=${TF_BINARIES_DIR}" \
+            --env "IREE_HOST_BINARY_ROOT=${BUILD_DIR}/install" \
+            gcr.io/iree-oss/base@sha256:5d43683c6b50aebe1fca6c85f2012f3b0fa153bf4dd268e8767b619b1891423a \
+            build_tools/cmake/build_benchmarks.sh
+
+  build_benchmark_tools:
+    runs-on:
+      # Hacks, and order matters. See the comment at the top of the ci.yml.
+      - self-hosted
```
Comment (GMNGeoffrey) [inline .github/workflows/benchmarks.yml:84]: Ditto comments above

```suggestion
  build_tools:
    runs-on:
      - self-hosted  # must come first
```

**Review by GMNGeoffrey — APPROVED** [review]

File: .github/workflows/benchmarks.yml (hunk 3 above)
Comment (pzread) [reply .github/workflows/benchmarks.yml]: Looks like there might be some deps to be cleaned up.
https://github.com/iree-org/iree/runs/8256083640

Comment (GMNGeoffrey) [reply .github/workflows/benchmarks.yml:63]: Darn. Thanks for checking

**Pull request merged**